### ISSUE TITLE
Automaticly updating the clangd LSP is not working

### ISSUE THREAD

**Issue by minego** [Civitasv/cmake-tools.nvim#182]
I enabled the clangd callback as explained in the docs, and have this in my config for cmake-tools: ```
cmake_soft_link_compile_commands	= false,
cmake_compile_commands_from_lsp		= true,
```

But, it doesn't appear to be working. When I run `:LspInfo` I see that clangd was started with this command:
```
clangd --background-index --limit-results=0 --compile-commands-dir=/home/m/src/venafi/csc/./build/default 
```

I cleared out the session files in ~/.config and I can't think of anything else that could be causing it to fail.

**Comment by Civitasv** [comment]
See docs: https://github.com/Civitasv/cmake-tools.nvim/blob/8d63e251b84e9a5d5c00b97a91983748c1c1cdb8/docs/howto.md?plain=1#L3

**Comment by minego** [comment]
I followed those instructions. I don't know what I might have done wrong...

**Comment by Civitasv** [comment]
Weird, I cannot reproduce this error. And this is my configuration for clangd: https://github.com/Civitasv/runvim/blob/19863e135f5b990bea1cb2d6cfab0b9776dc5918/lua/plugins/lsp/configs.lua#L88-L102.

**Comment by minego** [comment]
Thanks. I'll try to debug more tomorrow. I'm probably missing some small detail.

**Comment by minego** [comment]
Okay, I think I got this working now. I guess I was too tired yesterday, and missed some obvious bits. I had not included `-DCMAKE_EXPORT_COMPILE_COMMANDS=1` in my default config options because it used to be turned on by default for the project I'm working on. But, at some point that was changed and I hadn't realized it.

Anyway, I added that back today and it seems to be working now. Sorry for the confusion, and thanks for the help!
